### PR TITLE
Added new Python tutorial

### PR DESCRIPTION
**Pull request by marksou**
New Custom Authentication plugin tutorial for Python

### CODE REVIEW

File: tyk-docs/content/customise-tyk/plugins/rich-plugins/python/custom-auth-python-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+
+## <a name="next"></a>What's Next?
+
+In this tutorial we learned how Tyk plugins work. For a production-level setup we suggest the following steps:
+
+* Configure Tyk to use your own key so that you can enforce cryptographic signature checks when loading plugin bundles, and sign your plugin bundles!
+* Configure an appropriate web server and path to serve your plugin bundles.
+
+
```
Comment (buger) [inline tyk-docs/content/customise-tyk/plugins/rich-plugins/python/custom-auth-python-tutorial.md:163]: Why so many new lines?

Comment (marksou) [reply tyk-docs/content/customise-tyk/plugins/rich-plugins/python/custom-auth-python-tutorial.md:163]: It's a new page :-) I need to add some other bits Matias added to the doc, so will do another push.

File: tyk-docs/content/customise-tyk/plugins/rich-plugins/python/custom-auth-python-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+---
+
+## <a name="introduction"></a>Introduction
+This tutorial will guide you through the creation of a custom authentication plugin, written in Python.
+A custom authentication plugin allows you to implement your own authentication logic and override the default Tyk authentication mechanism. The sample code implements a very simple key check; currently it supports a single, hard-coded key. It could serve as a starting point for your own authentication logic. We have tested this plugin with Ubuntu 14.
+
+## <a name="requirements"></a>Requirements
+
+* Tyk API Gateway: This can be installed using standard package management tools like Yum or APT, or from source code. See [here][1] for more installation options.
+* Check you have the Tyk CLI installed: find `/opt/ | grep "tyk-cli"`. It should be found in `/opt/tyk-gateway/utils/tyk-cli`.
```
Comment (buger) [inline tyk-docs/content/customise-tyk/plugins/rich-plugins/python/custom-auth-python-tutorial.md]: I think this step can freak people out and it looks quite nerdy. Can be replaced with smth like:
```
Dependencies:
* Tyk CLI utility available from [https://github.com/TykTechnologies/tyk-cli][2]
* Python 3.4
```

Comment (buger) [reply tyk-docs/content/customise-tyk/plugins/rich-plugins/python/custom-auth-python-tutorial.md]: Actually, this is also not the best thing as well. Tyk CLI utility is bundled into our DEB and RPM installations, so should be installed only for the docker installation.

So we can re-phrase it smth like:
` Tyk CLI utility, which is bundled with our RPM and DEB packages, and can be installed separately from [https://github.com/TykTechnologies/tyk-cli][2]`

Comment (marksou) [reply tyk-docs/content/customise-tyk/plugins/rich-plugins/python/custom-auth-python-tutorial.md]: Changed and Pushed

Comment (marksou) [reply tyk-docs/content/customise-tyk/plugins/rich-plugins/python/custom-auth-python-tutorial.md]: Amended as per your comment

**Comment by buger** [comment]
👍

**Pull request merged**